Invalid type declaration after upgrading to TS 4.7

### [READ] Step 1: Are you in the right place?

  * For issues related to __the code in this repository__ file a Github issue.
  * If the issue pertains to Cloud Firestore, read the instructions in the "Firestore issue"
    template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Operating System version: MacOS Monterey 12.3.1
  * Firebase SDK version: 10.2.0
  * Firebase Product: app, firestore
  * Node.js version: 16
  * NPM version: 8.5.5

### [REQUIRED] Step 3: Describe the problem

#### Steps to reproduce:

After upgrading to Typescript 4.7, `tsc` throws the following error for all firebase admin SDK imports

```bash
error TS7016: Could not find a declaration file for module 'firebase-admin/firestore'. '/Users/debkanchan/Library/Mobile Documents/com~apple~CloudDocs/Work/Ride/Code/Backend/services/wallet-service/node_modules/firebase-admin/lib/esm/firestore/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/firebase-admin` if it exists or add a new declaration (.d.ts) file containing `declare module 'firebase-admin/firestore';`

2 import { Firestore, FieldValue, getFirestore } from "firebase-admin/firestore";
```

#### Relevant Code:
tsconfig.json

```jsonc
{
	"extends": "@tsconfig/node16/tsconfig.json",
	"compilerOptions": {
		"module": "Node16",
		"outDir": "build",
		"removeComments": true
		// "noImplicitAny": false
	},
	"compileOnSave": true,
	"include": ["src"]
}
```

package.json
```jsonc
{
	"type": "module",
	"engines": {
		"node": "16"
	},
       ...
	"dependencies": {
		"firebase-admin": "^10.2.0",
                ...
	},
	"devDependencies": {
		"@tsconfig/node16": "^1.0.2",
                 ...
		"ts-node": "^10.8.0",
		"typescript": "^4.7.2"
	}
}

```

--- a/package.json
+++ b/package.json
@@ -113,63 +113,78 @@
     ".": "./lib/index.js",
     "./app": {
       "require": "./lib/app/index.js",
-      "import": "./lib/esm/app/index.js"
+      "import": "./lib/esm/app/index.js",
+      "types": "./lib/app/index.d.ts"
     },
     "./app-check": {
       "require": "./lib/app-check/index.js",
-      "import": "./lib/esm/app-check/index.js"
+      "import": "./lib/esm/app-check/index.js",
+      "types": "./lib/app-check/index.d.ts"
     },
     "./auth": {
       "require": "./lib/auth/index.js",
-      "import": "./lib/esm/auth/index.js"
+      "import": "./lib/esm/auth/index.js",
+      "types": "./lib/auth/index.d.ts"
     },
     "./database": {
       "require": "./lib/database/index.js",
-      "import": "./lib/esm/database/index.js"
+      "import": "./lib/esm/database/index.js",
+      "types": "./lib/database/index.d.ts"
     },
     "./eventarc": {
       "require": "./lib/eventarc/index.js",
-      "import": "./lib/esm/eventarc/index.js"
+      "import": "./lib/esm/eventarc/index.js",
+      "types": "./lib/eventarc/index.d.ts"
     },
     "./firestore": {
       "require": "./lib/firestore/index.js",
-      "import": "./lib/esm/firestore/index.js"
+      "import": "./lib/esm/firestore/index.js",
+      "types": "./lib/firestore/index.d.ts"
     },
     "./functions": {
       "require": "./lib/functions/index.js",
-      "import": "./lib/esm/functions/index.js"
+      "import": "./lib/esm/functions/index.js",
+      "types": "./lib/functions/index.d.ts"
     },
     "./installations": {
       "require": "./lib/installations/index.js",
-      "import": "./lib/esm/installations/index.js"
+      "import": "./lib/esm/installations/index.js",
+      "types": "./lib/installations/index.d.ts"
     },
     "./instance-id": {
       "require": "./lib/instance-id/index.js",
-      "import": "./lib/esm/instance-id/index.js"
+      "import": "./lib/esm/instance-id/index.js",
+      "types": "./lib/instance-id/index.d.ts"
     },
     "./machine-learning": {
       "require": "./lib/machine-learning/index.js",
-      "import": "./lib/esm/machine-learning/index.js"
+      "import": "./lib/esm/machine-learning/index.js",
+      "types": "./lib/machine-learning/index.d.ts"
     },
     "./messaging": {
       "require": "./lib/messaging/index.js",
-      "import": "./lib/esm/messaging/index.js"
+      "import": "./lib/esm/messaging/index.js",
+      "types": "./lib/messaging/index.d.ts"
     },
     "./project-management": {
       "require": "./lib/project-management/index.js",
-      "import": "./lib/esm/project-management/index.js"
+      "import": "./lib/esm/project-management/index.js",
+      "types": "./lib/project-management/index.d.ts"
     },
     "./remote-config": {
       "require": "./lib/remote-config/index.js",
-      "import": "./lib/esm/remote-config/index.js"
+      "import": "./lib/esm/remote-config/index.js",
+      "types": "./lib/remote-config/index.d.ts"
     },
     "./security-rules": {
       "require": "./lib/security-rules/index.js",
-      "import": "./lib/esm/security-rules/index.js"
+      "import": "./lib/esm/security-rules/index.js",
+      "types": "./lib/security-rules/index.d.ts"
     },
     "./storage": {
       "require": "./lib/storage/index.js",
-      "import": "./lib/esm/storage/index.js"
+      "import": "./lib/esm/storage/index.js",
+      "types": "./lib/storage/index.d.ts"
     }
   },
   "dependencies": {

--- a/package.json
+++ b/package.json
@@ -112,79 +112,79 @@
   "exports": {
     ".": "./lib/index.js",
     "./app": {
+      "types": "./lib/app/index.d.ts",
       "require": "./lib/app/index.js",
-      "import": "./lib/esm/app/index.js",
-      "types": "./lib/app/index.d.ts"
+      "import": "./lib/esm/app/index.js"
     },
     "./app-check": {
+      "types": "./lib/app-check/index.d.ts",
       "require": "./lib/app-check/index.js",
-      "import": "./lib/esm/app-check/index.js",
-      "types": "./lib/app-check/index.d.ts"
+      "import": "./lib/esm/app-check/index.js"
     },
     "./auth": {
+      "types": "./lib/auth/index.d.ts",
       "require": "./lib/auth/index.js",
-      "import": "./lib/esm/auth/index.js",
-      "types": "./lib/auth/index.d.ts"
+      "import": "./lib/esm/auth/index.js"
     },
     "./database": {
+      "types": "./lib/database/index.d.ts",
       "require": "./lib/database/index.js",
-      "import": "./lib/esm/database/index.js",
-      "types": "./lib/database/index.d.ts"
+      "import": "./lib/esm/database/index.js"
     },
     "./eventarc": {
+      "types": "./lib/eventarc/index.d.ts",
       "require": "./lib/eventarc/index.js",
-      "import": "./lib/esm/eventarc/index.js",
-      "types": "./lib/eventarc/index.d.ts"
+      "import": "./lib/esm/eventarc/index.js"
     },
     "./firestore": {
+      "types": "./lib/firestore/index.d.ts",
       "require": "./lib/firestore/index.js",
-      "import": "./lib/esm/firestore/index.js",
-      "types": "./lib/firestore/index.d.ts"
+      "import": "./lib/esm/firestore/index.js"
     },
     "./functions": {
+      "types": "./lib/functions/index.d.ts",
       "require": "./lib/functions/index.js",
-      "import": "./lib/esm/functions/index.js",
-      "types": "./lib/functions/index.d.ts"
+      "import": "./lib/esm/functions/index.js"
     },
     "./installations": {
+      "types": "./lib/installations/index.d.ts",
       "require": "./lib/installations/index.js",
-      "import": "./lib/esm/installations/index.js",
-      "types": "./lib/installations/index.d.ts"
+      "import": "./lib/esm/installations/index.js"
     },
     "./instance-id": {
+      "types": "./lib/instance-id/index.d.ts",
       "require": "./lib/instance-id/index.js",
-      "import": "./lib/esm/instance-id/index.js",
-      "types": "./lib/instance-id/index.d.ts"
+      "import": "./lib/esm/instance-id/index.js"
     },
     "./machine-learning": {
+      "types": "./lib/machine-learning/index.d.ts",
       "require": "./lib/machine-learning/index.js",
-      "import": "./lib/esm/machine-learning/index.js",
-      "types": "./lib/machine-learning/index.d.ts"
+      "import": "./lib/esm/machine-learning/index.js"
     },
     "./messaging": {
+      "types": "./lib/messaging/index.d.ts",
       "require": "./lib/messaging/index.js",
-      "import": "./lib/esm/messaging/index.js",
-      "types": "./lib/messaging/index.d.ts"
+      "import": "./lib/esm/messaging/index.js"
     },
     "./project-management": {
+      "types": "./lib/project-management/index.d.ts",
       "require": "./lib/project-management/index.js",
-      "import": "./lib/esm/project-management/index.js",
-      "types": "./lib/project-management/index.d.ts"
+      "import": "./lib/esm/project-management/index.js"
     },
     "./remote-config": {
+      "types": "./lib/remote-config/index.d.ts",
       "require": "./lib/remote-config/index.js",
-      "import": "./lib/esm/remote-config/index.js",
-      "types": "./lib/remote-config/index.d.ts"
+      "import": "./lib/esm/remote-config/index.js"
     },
     "./security-rules": {
+      "types": "./lib/security-rules/index.d.ts",
       "require": "./lib/security-rules/index.js",
-      "import": "./lib/esm/security-rules/index.js",
-      "types": "./lib/security-rules/index.d.ts"
+      "import": "./lib/esm/security-rules/index.js"
     },
     "./storage": {
+      "types": "./lib/storage/index.d.ts",
       "require": "./lib/storage/index.js",
-      "import": "./lib/esm/storage/index.js",
-      "types": "./lib/storage/index.d.ts"
+      "import": "./lib/esm/storage/index.js"
     }
   },
   "dependencies": {

